internal/engine/configs_controller.go

```diff
@@ -27,23 +28,44 @@ func (cc *ConfigsController) DisableForTesting(disabled bool) {
 	cc.disabledForTesting = disabled
 }
 
+// Modeled after BuildController.nextTargetToBuild. Check to see that:
+// 1) There's currently no Tiltfile build running,
+// 2) There are pending file changes, and
+// 3) Those files have changed since the last Tiltfile build
+//    (so that we don't keep re-running a failed build)
+func (cc *ConfigsController) hasChangesToBuild(state store.EngineState) bool {
+	isRunning := !state.CurrentTiltfileBuild.StartTime.IsZero()
```

Comment:
naming nit: "has changes to build" isn't quite the same as "should start a build" (i.e., there could be changes to build but we choose not to build because another build is already running)

maybe `shouldBuild`?

Reply:
done!